daemon/entrypoint.sh

```diff
@@ -188,12 +188,26 @@ function start_osd {
 #################
 
 function osd_directory {
-  if [[ ! -d /var/lib/ceph/osd || -n "$(find /var/lib/ceph/osd -prune -empty)" ]]; then
-    echo "ERROR- could not find any OSD, did you bind mount the OSD data directory?"
-    echo "ERROR- use -v <host_osd_data_dir>:<container_osd_data_dir>"
+  if [[ ! -d /var/lib/ceph/osd ]]; then
+    echo "ERROR- could not find the osd directory, did you bind mount the OSD data directory?"
+    echo "ERROR- use -v <host_osd_data_dir>:/var/lib/ceph/osd"
     exit 1
   fi
 
+  # make sure ceph owns the directory
+  chown ceph. /var/lib/ceph/osd
```

Comment:
This should probably be a recursive chown, and I don't see a reason not to set the group, as well (though perhaps that is too invasive):  `chown -R ceph:ceph /var/lib/ceph/osd`